crates/aptos/src/common/init.rs

```diff
@@ -154,7 +158,25 @@ impl CliCommand<()> for InitTool {
             }
         };
         let public_key = private_key.public_key();
-        let address = account_address_from_public_key(&public_key);
+
+        // lookup the address from onchain instead of deriving it
+        // if this is the rotated key, deriving it will outputs an incorrect address
+        let address = if let Some(rest_url) = &profile_config.rest_url {
+            LookupAddress {
+                encoding_options: Default::default(),
+                public_key_options: PublicKeyInputOptions::from_key(&public_key),
+                profile_options: Default::default(),
+                rest_options: RestOptions::new(
+                    Option::from(Url::parse(rest_url).expect("Failed to parse url")),
+                    None,
+                ),
+            }
+            .execute()
+            .await?
+        } else {
+            account_address_from_public_key(&public_key)
```

Comment:
@gregnazario , we also have an on-chain repository of rotated account address -> actual account address: https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/account.move#L62

it might be worth adding this into the cli as well.

Reply:
Looking at the `OriginatingAddress` struct. This is for looking up the originating address of the rotated auth key?

I guess we can add something like this in the CLI? -> `aptos account lookup-address --authentication-key`